src/consensus/sync_queue.cpp

```diff
@@ -0,0 +1,44 @@
+#include "sync_queue.hpp"
+
+#include "consensus/pbft_chain.hpp"
+#include "consensus/sync_block.hpp"
+#include "dag/dag_block.hpp"
+#include "transaction_manager/transaction.hpp"
+
+namespace taraxa {
+
+uint64_t SyncBlockQueue::getPeriod() const {
+  std::shared_lock lock(sync_queue_access_);
+  return sync_period_;
+}
+
+size_t SyncBlockQueue::size() const {
+  std::shared_lock lock(sync_queue_access_);
+  return sync_queue_.size();
+}
+
+void SyncBlockQueue::clear() {
+  std::unique_lock lock(sync_queue_access_);
+  sync_period_ = 0;
+  sync_queue_.clear();
+}
+
+bool SyncBlockQueue::push(SyncBlock &&block, dev::p2p::NodeID const &node_id, uint64_t max_pbft_size) {
+  const auto period = block.pbft_blk->getPeriod();
+  std::unique_lock lock(sync_queue_access_);
+  if (period != std::max(sync_period_, max_pbft_size) + 1) {
+    return false;
+  }
+  sync_period_ = period;
+  sync_queue_.emplace_back(std::move(block), node_id);
+  return true;
+}
+
+std::pair<SyncBlock, dev::p2p::NodeID> SyncBlockQueue::pop() {
+  std::unique_lock lock(sync_queue_access_);
+  auto sync_block = std::move(sync_queue_.front());
+  sync_queue_.pop_front();
```

Comment:
I think you should check here if sync_queue_ is empty and if so, set sync_period_to 0

Reply:
hmm or maybe not, we cant do it because we need to keep that value for the time between popping from queue and pushing into the chain right ?

Reply:
nope, because of processing of this block can take a while, we can to set it to 0

Reply:
That's the reason why we have the sync_period_ there